### DISCUSSION TITLE
Feature request: supports for stores in the defineDisplay function

### DISCUSSION THREAD

**Discussion by mattiaz9** [directus/directus#10255] [Feature Requests]
As stated in this bug https://github.com/directus/directus/issues/9802 the `fields` function in `defineDisplay` should support the usage of the stores. The reason behind this is if a core display can uses the stores, so should a custom display, whether is to create a whole new extension, or to extend a core one.

Since, apparently, `useStores` cannot work outside the Vue component scope, my idea to make this work is to provide a property `stores` in the `context` object of the fields function.

This should be fairly easy to implement (I could open I pull request myself), but I'm not sure if the various `useFieldsStore`, `useRelationsStore` ext... could work or they would have the same problem of the `useStores` function. 

If this is indeed possibile, then another point of this discussion should be if the stores property should also be provided in the `options` function as well to make it more consistent.

**Comment by rijkvanzanten** [comment]
I believe `useStores` isn't accessible there, because it has to rely on provide/inject to get the stores. If we pass them through as the function parameter, things might work the way you'd expect them to.

Any thoughts @nickrum?

**Reply by nickrum** [reply]
Yeah, `fields()` and `options()` are executed inside computed functions where the Vue instance isn't available. Pinia does some magic so the store composables are also available when there is no current Vue instance, so just passing the stores to those functions should work. It feels a little verbose, though. If we add additional functions, we would have to pass the stores to them as well.

**Reply by AxeemHaider** [reply]
So, then how we can use it inside `definePanel` i want to use `useCollectionsStore()` in `definePanel->Query`.
How can I use it there?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
